agent/agent.go

```diff
@@ -4027,17 +4032,18 @@ func (a *Agent) reloadConfig(autoReload bool) error {
 			{a.config.TLS.HTTPS, newCfg.TLS.HTTPS},
 		} {
 			if f.oldCfg.KeyFile != f.newCfg.KeyFile {
-				a.configFileWatcher.Replace(f.oldCfg.KeyFile, f.newCfg.KeyFile)
+				err = a.configFileWatcher.Replace(f.oldCfg.KeyFile, f.newCfg.KeyFile)
 				if err != nil {
 					return err
 				}
 			}
 			if f.oldCfg.CertFile != f.newCfg.CertFile {
-				a.configFileWatcher.Replace(f.oldCfg.CertFile, f.newCfg.CertFile)
+				err = a.configFileWatcher.Replace(f.oldCfg.CertFile, f.newCfg.CertFile)
 				if err != nil {
 					return err
 				}
 			}
+
```

Comment:
I added this because currently the statements below will not return errors if the configFileWatcher swap returns an error.  I can move this to an outside PR, but either this change needs to occur or we just remove the error returns below them because they are not actually doing the way this is currently written.

Reply:
agree. will do

Reply:
This is now in https://github.com/hashicorp/consul/pull/15769.  It's currently in draft mode.  I'll fix up the description and changelog.